### PR TITLE
feat: improve balances query

### PR DESCRIPTION
**Pull request by ratik**
### this PR:
- deprecates the SDK's `new_register_balance_query_msg` helper
- adds a new `new_register_balances_query_msg` that accepts a list of denoms to query balances of
- fixes v045 `Balance` reconstruction for empty values

### related PRs:
- https://github.com/neutron-org/neutron-dev-contracts/pull/41
- https://github.com/neutron-org/neutron-integration-tests/pull/298

### CODE REVIEW

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
Let's create a separate method for this, not replacing the old one, cause otherwise the change is not backwards compatible

**Comment by pr0n00gler** [comment]
And let's mark it as deprecated, please

File: packages/neutron-sdk/src/interchain_queries/v045/register_queries.rs [hunk 1]
```diff
@@ -17,32 +17,50 @@ use crate::{
...
-        connection_id,
-        update_period,
-    )
+        let kv_key = KVKey {
+            path: BANK_STORE_KEY.to_string(),
+            key: Binary(balance_key),
+        };
+
+        kv_keys.push(kv_key)
+    }
```
Comment (albertandrejev) [inline packages/neutron-sdk/src/interchain_queries/v045/register_queries.rs:45]: better move this part into separate helper, it will help to create kv_keys in case if you want to update interchain query.

Comment (ratik) [reply packages/neutron-sdk/src/interchain_queries/v045/register_queries.rs]: sure, thing! 👍

**Review by albertandrejev — APPROVED** [review]

**Review by pr0n00gler — COMMENTED** [review]
Please resolve conflicts

**Review by albertandrejev — APPROVED** [review]

**Review by pr0n00gler — DISMISSED** [review]

**Comment by ratik** [comment]
image
image

**Review by albertandrejev — APPROVED** [review]

**Comment by ratik** [comment]
[Test run](https://github.com/neutron-org/neutron-tests/actions/runs/8038720349/job/21956680355)

**Review by sotnikov-s — CHANGES_REQUESTED** [review]
I think we need to

1. extend ICQ contract's interface to accept a list of denoms for balances query
2. add integration tests covering multiple denoms and response with a zero value balance

will handle it myself later

**Comment by sotnikov-s** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/9320148190

**Review by sotnikov-s — DISMISSED** [review]

**Review by NeverHappened — DISMISSED** [review]

File: packages/neutron-sdk/src/interchain_queries/v045/helpers.rs [hunk 2]
```diff
@@ -46,6 +46,27 @@ pub fn create_account_denom_balance_key<AddrBytes: AsRef<[u8]>, S: AsRef<str>>(
...
+/// Creates keys for an Interchain Query to get balance of account on remote chain for list of denoms
+///
+/// * **addr** address of an account on remote chain for which you want to get balances;
+/// * **denoms** denominations of the coins for which you want to get balance;
+pub fn create_balances_query_keys(addr: String, denoms: Vec<String>) -> NeutronResult<Vec<KVKey>> {
+    let converted_addr_bytes = decode_and_convert(addr.as_str())?;
+    let mut kv_keys: Vec<KVKey> = Vec::with_capacity(denoms.len());
+
+    for denom in denoms {
+        let balance_key = create_account_denom_balance_key(converted_addr_bytes.clone(), denom)?;
```
Comment (pr0n00gler) [inline packages/neutron-sdk/src/interchain_queries/v045/helpers.rs:58]:
```suggestion
        let balance_key = create_account_denom_balance_key(&converted_addr_bytes, denom)?;
```

**Review by pr0n00gler — CHANGES_REQUESTED** [review]

**Review by sotnikov-s — APPROVED** [review]

**Review by NeverHappened — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]
